### PR TITLE
Make ServerState.shutting_down into an enum

### PR DESCRIPTION
**Pull request by spladug**
This is more extensible and should've been done all along.

### CODE REVIEW

**Comment by spladug** [comment]
:eyeglasses: @mattknox

File: baseplate/server/__init__.py [hunk 1]
```diff
@@ -47,9 +50,19 @@
 logger = logging.getLogger(__name__)
 
 
+class ServerLifecycle(Enum):
+    RUNNING = enum.auto()
+    SHUTTING_DOWN = enum.auto()
```
Comment (mattknox) [inline baseplate/server/__init__.py:55]: This is great!  Is the ordering sensitive?  (It almost certainly doesn't matter unless you're serializing stuff, which is hard to imagine)

File: baseplate/server/__init__.py [hunk 2]
```diff
@@ -266,7 +279,7 @@ def load_app_and_run_server() -> None:
     try:
         shutdown_event.wait()
 
-        SERVER_STATE.shutting_down = True
+        SERVER_STATE.state = ServerLifecycle.SHUTTING_DOWN
```
Comment (mattknox) [inline baseplate/server/__init__.py:282]: Lovely!

**Review by mattknox — APPROVED** [review]
My name is Matt Knox, and I approve this change.

File: baseplate/server/__init__.py (hunk 1 above)
Comment (spladug) [reply baseplate/server/__init__.py:55]: yeah, it's ordering sensitive. if we added a new one inbetween it'd mess up the latter but this state should only ever make sense in-process

**Pull request merged**